recipes/quantize.py

```diff
@@ -93,7 +104,10 @@ def _setup_model(
     @torch.no_grad()
     def quantize(self, cfg: DictConfig):
         t0 = time.perf_counter()
-        self._model = self._quantizer.quantize(self._model)
+        if "qat" in self._quantization_mode:
+            self._model = self._quantizer.convert(self._model)
+        else:
+            self._model = self._quantizer.quantize(self._model)
```

Comment:
So we need to gate on quantization mode because the QAT checkpoints are in bf16, right? Noob question but if we are not doing any subsequent training why can't we just call `.quantize` directly and infer all the quantizer params from the checkpoint?

Reply:
Technically we can, the numerics may be the the same, but officially the torchao QAT flow is:

```
quantizer = 8da4wQATQuantizer()
model = quantizer.prepare(model)
train(model)
model = quantizer.convert(model)
```

If we just call quantize here we would have to introduce a different quantizer

```
quantizer = 8da4wQATQuantizer()
model = quantizer.prepare(model)
train(model)
ptq_quantizer = 8da4wQuantizer()
ptq_quantizer.quantize(model)
```

I feel it's better to call the complete QAT flow rather than to switch quantizers in the middle